### DISCUSSION TITLE
(rows per page)page size set to 1000 is affecting table performance

### DISCUSSION THREAD

**Discussion by rmshmaudlin** [KevinVandy/material-react-table#1164] [Q&A]
Table is too slow(takes more than 10 seconds to load the table) if I initialize pageSize with 1000. 
If the page size is 100 or less it works perfectly. My data contains atleast 25 columns.
Is there any solution to load faster when pageSize set to 1000 or more.

**Comment by KevinVandy** [comment]
Well that makes sense. That's a lot of dom nodes, right?

**Reply by rmshmaudlin** [reply]
@KevinVandy  Do you have any work around or possible solution?

**Reply by KevinVandy** [reply]
Pagination and virtualization features

**Reply by rmshmaudlin** [reply]
I don't want virtualization feature.
 I need pagination with 1000 rows per page with decent loading time.

**Reply by KevinVandy** [reply]
Well, you're asking for something that's impossible then. Or you are about to work hard and create the best new table with the best performance of all time and become very famous.

But seriously, that's why the virtualization features exist. It's the only way to try to make loading thousands of client side rows in the same page at the same time possible and somewhat performant.

**Reply by KevinVandy** [reply]
Though also, if you don't need a lot of the fancy MRT features, and your table needs to focus on SEO and making sure that all of the data is just displayed as efficiently as possible, MRT might not be the best choice for that. Consider just rendering out the markup for a table like that manually with either the MRT headless mode, stock/vanilla TanStack Table, or your own lightweight solution.

**Reply by rmshmaudlin** [reply]
I'm using most of the MRT features. Really awesome table. BTW Thanks for the reply

**Comment by rmshmaudlin** [comment]
@KevinVandy Column size is not autofit by content with virtualization enabled
